### PR TITLE
Persisting containers and added helper functions for building pod-pod, service-service interactions.

### PR DESCRIPTION
**Pull request by gurusreekanthc**
Implemented the following.
- Persisting containers and its relation to Pod.
 - Helper functions for building interaction graphs for pod-to-pod
   and service-to-service communication.

### CODE REVIEW

**Comment by kreddyj** [comment]
@gurusreekanthc build is failing as there are two main functions in the same package. https://travis-ci.org/vmware/purser/builds/439312202?utm_source=github_status&utm_medium=notification

File: pkg/controller/persistence/dgraph/service.go [hunk 1]
```diff
@@ -1,37 +1,99 @@
...
+	}
+
+	services := []*Service{}
+	for _, destinationService := range destinationServices {
+		uid, err := GetUId(Client, destinationService, IsService)
+		if err != nil {
+			return err
+		}
+
+		if uid == "" {
```
Comment (kreddyj) [inline pkg/controller/persistence/dgraph/service.go]: WIP if uid is ""?

File: pkg/controller/persistence/dgraph/service.go [hunk 2]
```diff
@@ -1,37 +1,99 @@
...
+			continue
+		}
+
+		service := &Service{
+			ID: ID{UID: uid},
+		}
+		services = append(services, service)
+	}
+	source := Service{
+		ID:        ID{UID: uid},
```
Comment (kreddyj) [inline pkg/controller/persistence/dgraph/service.go]: WIP if source UID is ""?

File: pkg/controller/persistence/dgraph/pod.go [hunk 3]
```diff
@@ -14,38 +38,128 @@ type ID struct {
...
+
+	pods := []*Pod{}
+	for _, destinationPod := range destinationPods {
+		uid, err := GetUId(Client, destinationPod, IsPod)
 		if err != nil {
 			return err
 		}
-		return MutateNode(Client, bytes)
+
+		if uid == "" {
```
Comment (kreddyj) [inline pkg/controller/persistence/dgraph/pod.go]: WIP if uid is ""?

File: pkg/controller/persistence/dgraph/pod.go [hunk 4]
```diff
@@ -14,38 +38,128 @@ type ID struct {
...
+			continue
+		}
+
+		pod := &Pod{
+			ID: ID{UID: uid},
+		}
+		pods = append(pods, pod)
 	}
-	return nil
+	source := Pod{
```
Comment (kreddyj) [inline pkg/controller/persistence/dgraph/pod.go]: WIP if source pod uid is ""?

File: pkg/controller/persistence/dgraph/pod.go [hunk 5]
```diff
@@ -14,38 +38,128 @@ type ID struct {
 
 type Pod struct {
 	ID
-	IsPod      string       `json:"isPod"`
+	IsPod      bool         `json:"isPod,omitempty"`
 	Name       string       `json:"name,omitempty"`
 	StartTime  time.Time    `json:"startTime,omitempty"`
 	EndTime    time.Time    `json:"endTime,omitempty"`
 	Containers []*Container `json:"containers,omitempty"`
+	Interacts  []*Pod       `json:interacts,omitempty`
```
Comment (kreddyj) [inline pkg/controller/persistence/dgraph/pod.go]: Missing quotes ``json:"interacts,omitempty"``

File: cmd/controller/main.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
```
[the remark this answers is not in the document]
Comment (gurusreekanthc) [reply cmd/controller/main.go]: renamed main function in main.go as it is a temporary file.

File: pkg/controller/persistence/dgraph/service.go (hunk 1 above)
Comment (gurusreekanthc) [reply pkg/controller/persistence/dgraph/service.go]: If service is not already persisted then we don't update "interacts" edge.

File: pkg/controller/persistence/dgraph/pod.go (hunk 3 above)
Comment (gurusreekanthc) [reply pkg/controller/persistence/dgraph/pod.go]: If pod is not already persisted then we don't update "interacts" edge.

File: pkg/controller/persistence/dgraph/pod.go (hunk 4 above)
Comment (gurusreekanthc) [reply pkg/controller/persistence/dgraph/pod.go]: returning from function if UID is nil.

File: pkg/controller/eventprocessor/processor.go [hunk 7]
```diff
@@ -40,9 +40,6 @@ func ProcessEvents(conf *config.Config) {
 			// Update user created groups.
 			controller.UpdateCustomGroups(data, groups, conf.Groupcrdclient)
 
-			// Persist in dgraph
-			PersistPayloads(data)
```
[the remark this answers is not in the document]
Comment (gurusreekanthc) [reply pkg/controller/eventprocessor/processor.go]: We will enable once things are stable.

File: pkg/controller/persistence/dgraph/pod.go (hunk 5 above)
Comment (gurusreekanthc) [reply pkg/controller/persistence/dgraph/pod.go]: good catch.

**Comment by kreddyj** [comment]
@gurusreekanthc LGTM :+1:

**Pull request merged**